aptos-move/aptos-debugger/src/execute_pending_block.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{aptos_debugger::AptosDebugger, common::Opts};
+use anyhow::Result;
+use aptos_crypto::HashValue;
+use aptos_logger::info;
+use aptos_rest_client::Client;
+use aptos_vm::AptosVM;
+use clap::Parser;
+use std::path::PathBuf;
+use url::Url;
+
+#[derive(Parser)]
+#[clap(group(clap::ArgGroup::new("block_target")
+        .required(true)
+        .multiple(false)
+        .args(&["block_rest_endpoint", "consensus_db_path"]),
+))]
+pub struct Command {
+    #[clap(flatten)]
+    opts: Opts,
+
+    #[clap(long, group = "block_target")]
+    block_rest_endpoint: Option<String>,
+
+    #[clap(long, group = "block_target")]
+    consensus_db_path: Option<PathBuf>,
+
+    #[clap(long)]
+    begin_version: u64,
+
+    #[clap(long)]
+    block_id: Option<HashValue>,
+
+    #[clap(long)]
+    add_system_txns: bool,
+}
+
+impl Command {
+    pub async fn run(self) -> Result<()> {
+        AptosVM::set_concurrency_level_once(self.opts.concurrency_level);
+
+        let debugger = if let Some(rest_endpoint) = self.opts.target.rest_endpoint {
+            AptosDebugger::rest_client(Client::new(Url::parse(&rest_endpoint)?))?
+        } else if let Some(db_path) = self.opts.target.db_path {
+            AptosDebugger::db(db_path)?
+        } else {
+            unreachable!("Must provide one target.");
+        };
+
+        let user_txns = if let Some(block_rest_endpoint) = self.block_rest_endpoint {
+            info!(
+                "Getting block {:?} from {block_rest_endpoint:?}.",
```

Comment:
what is this {block_reset_endpoint:?}?

Reply:
The endpoint in AdminService to get txns in consensus_db/quorumstore_db by block id.